docs/devel/luns.md

```diff
@@ -0,0 +1,82 @@
+Direct LUNs migration
+==============
+
+Summary
+-------
+
+VM contains disks. Some of them might be connected as a direct LUN.
+When a user wish to re-use the LUN disks, he needs them attached to the VM.
+The LUN should be used in a single place, or else, it will cause a data corruption.
+The storage where the LUNs reside, needs to be reachable to the VM environment.
+This feature intend to support oVirt only.
+
+Design
+------
+
+The requirements for the LUN migration code are (not in priority order)
+
+1. Gather the LUN information from the source destination.
+2. To ease on the user - less manual actions, more automated migration.
+3. Avoid unnecessary risks.
+
+Requirements
+----------------------------------------
+
+1. Destination (OCP cluster) reachability to the ISCSI storage, where the LUN is.
+2. RBAC permission to create `PersistentVolumes` by the controller.
+   - Will be added to the operator manifest.
+
+Risks and Mitigations
+----------------------------------------
+
+1. A VM that will be used while or after the migration with a direct LUN connected.
+   - A warning should be shown before starting the migration.
+   - Detaching the LUN from source VM, once the migration completes.
+   - (Optional) UI checkbox to the user, before starting the migration.
+   - The migration _must_ be a cold migration.
+2. An oVirt version with unsupported API. The [PR](https://github.com/oVirt/ovirt-engine/pull/559) provides LUN details is required.
+   - We may check for missing LUN details and skip the disk migration.
+   - (Optional) Have a mechanism to check the oVirt version for future work.
+3. Creating a `PersistentVolume` requires the file system type: `ext4` or `xfs` of the LUN, which we don't know.
+   - Might not be required for `Block` type.
+4. On non-local destination, the controller needs permission to create the `PersistentVolumes`.
+   - It may be enough to be added to the forklift-operator.
+
+Implementation
+----------------------------------------
+
+The controller gather data from source provider and saves them to the inventory.
+As part of the data gathering we will gather the LUN disks data. We have the VMs disk attachments where we follow to the disks.
+Within the disks API, if the _storage\_type_ is `LUN`, we will gather from the API endpoint `api/disks/<disk_id>` the size, address, IQN and the mapping of the LUN.
+Those will be found withing the `logical_unit` element of the `lun_storage`.
+On the migration plan a new steps will be added, creating a `PersistentVolume`(PV) and a `PersistentVolumeClaim`(PVC) per LUN disk.
```

Comment:
why do we need to create a step for creating a PVC (as opposed to posting it on the VM itself)?

Reply:
you mean that the PVC will be created based on the VM spec? if so, how it will know which PV to bound?

Reply:
> you mean that the PVC will be created based on the VM spec?

yes, the pvc would be part of the vm specification (as the disks are today) and the pvc would be derived from that

> if so, how it will know which PV to bound?

the same way as you planned - by configuring the selection to match the pv

Reply:
I need to test it but sure, if it's possible.. why not.